### PR TITLE
[CF-149] Fix for tenant filtering

### PR DESCRIPTION
**Pull request by abochkarev**
This fix allows to avoid situations when cached method
'get_tenants_list' doesn't consider filtering by tenant.

### CODE REVIEW

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module cloudferrylib.os.image.glance_image
cloudferrylib/os/image/glance_image.py:412: [W0221(arguments-differ), GlanceImage.deploy] Arguments number differs from overridden 'deploy' method
************* Module cloudferrylib.os.identity.keystone
cloudferrylib/os/identity/keystone.py:197: [W0221(arguments-differ), KeystoneIdentity.deploy] Arguments number differs from overridden 'deploy' method
************* Module cloudferrylib.os.network.neutron
cloudferrylib/os/network/neutron.py:187: [W0221(arguments-differ), NeutronNetwork.deploy] Arguments number differs from overridden 'deploy' method
cloudferrylib/os/network/neutron.py:292: [W0221(arguments-differ), NeutronNetwork.create_port] Arguments number differs from overridden 'create_port' method
cloudferrylib/os/network/neutron.py:821: [W0221(arguments-differ), NeutronNetwork.get_security_groups] Arguments number differs from overridden 'get_security_groups' method

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by abochkarev** [comment]
retest this please

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module cloudferrylib.os.image.glance_image
cloudferrylib/os/image/glance_image.py:412: [W0221(arguments-differ), GlanceImage.deploy] Arguments number differs from overridden 'deploy' method
************* Module cloudferrylib.os.identity.keystone
cloudferrylib/os/identity/keystone.py:197: [W0221(arguments-differ), KeystoneIdentity.deploy] Arguments number differs from overridden 'deploy' method
************* Module cloudferrylib.os.network.neutron
cloudferrylib/os/network/neutron.py:187: [W0221(arguments-differ), NeutronNetwork.deploy] Arguments number differs from overridden 'deploy' method
cloudferrylib/os/network/neutron.py:292: [W0221(arguments-differ), NeutronNetwork.create_port] Arguments number differs from overridden 'create_port' method
cloudferrylib/os/network/neutron.py:821: [W0221(arguments-differ), NeutronNetwork.get_security_groups] Arguments number differs from overridden 'get_security_groups' method

**Comment by abochkarev** [comment]
retest this please

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module cloudferrylib.os.image.glance_image
cloudferrylib/os/image/glance_image.py:412: [W0221(arguments-differ), GlanceImage.deploy] Arguments number differs from overridden 'deploy' method
************* Module cloudferrylib.os.identity.keystone
cloudferrylib/os/identity/keystone.py:197: [W0221(arguments-differ), KeystoneIdentity.deploy] Arguments number differs from overridden 'deploy' method
************* Module cloudferrylib.os.network.neutron
cloudferrylib/os/network/neutron.py:187: [W0221(arguments-differ), NeutronNetwork.deploy] Arguments number differs from overridden 'deploy' method
cloudferrylib/os/network/neutron.py:292: [W0221(arguments-differ), NeutronNetwork.create_port] Arguments number differs from overridden 'create_port' method
cloudferrylib/os/network/neutron.py:821: [W0221(arguments-differ), NeutronNetwork.get_security_groups] Arguments number differs from overridden 'get_security_groups' method

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by khandelwalkk** [comment]
+1 works in staging and ready to merge

File: cloudferrylib/os/identity/keystone.py [hunk 1]
```diff
@@ -164,7 +164,7 @@ def read_info(self, **kwargs):
         if kwargs.get('tenant_id'):
             self.filter_tenant_id = kwargs['tenant_id'][0]
 
-        tenant_list = self.get_tenants_list()
+        tenant_list = self.get_tenants_list(self.filter_tenant_id)
```
Comment (roman-verchikov) [inline cloudferrylib/os/identity/keystone.py]: Passing instance member (`self.filter_tenant_id`) as argument to instance method (`self.get_tenants_list`) looks VERY strange. This smells bad code, please refactor.

**Pull request closed without merging**